internal/coreinternal/golden/normalize_timestamps.go

```diff
@@ -0,0 +1,250 @@
+				switch m.Type() {
+				case pmetric.MetricTypeGauge:
+					normGaugeDataPointSlice(m.Gauge().DataPoints())
+				case pmetric.MetricTypeSum:
+					normGaugeDataPointSlice(m.Sum().DataPoints())
+				case pmetric.MetricTypeHistogram:
+					normHistogramDataPointSlice(m.Histogram().DataPoints())
+				case pmetric.MetricTypeExponentialHistogram:
+					normExponentialHistogramDataPointSlice(m.ExponentialHistogram().DataPoints())
+				case pmetric.MetricTypeSummary:
+					normSummaryDataPointSlice(m.Summary().DataPoints())
+				}
+			}
+		}
+	}
+}
+
+func zeroStartTimestamp(metrics pmetric.Metrics) {
+	zeroTime := time.Unix(0, 0)
+	nanosec := pcommon.NewTimestampFromTime(zeroTime)
+	rms := metrics.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		for j := 0; j < rms.At(i).ScopeMetrics().Len(); j++ {
+			for k := 0; k < rms.At(i).ScopeMetrics().At(j).Metrics().Len(); k++ {
+				m := rms.At(i).ScopeMetrics().At(j).Metrics().At(k)
+				switch m.Type() {
+				case pmetric.MetricTypeGauge:
+					for l := 0; l < m.Gauge().DataPoints().Len(); l++ {
+						m.Gauge().DataPoints().At(l).SetStartTimestamp(nanosec)
+					}
+				case pmetric.MetricTypeSum:
+					for l := 0; l < m.Sum().DataPoints().Len(); l++ {
+						m.Sum().DataPoints().At(l).SetStartTimestamp(nanosec)
+					}
+				case pmetric.MetricTypeHistogram:
+					for l := 0; l < m.Histogram().DataPoints().Len(); l++ {
+						m.Histogram().DataPoints().At(l).SetStartTimestamp(nanosec)
+					}
+				case pmetric.MetricTypeExponentialHistogram:
+					for l := 0; l < m.ExponentialHistogram().DataPoints().Len(); l++ {
+						m.ExponentialHistogram().DataPoints().At(l).SetStartTimestamp(nanosec)
+					}
+				case pmetric.MetricTypeSummary:
+					for l := 0; l < m.Summary().DataPoints().Len(); l++ {
+						m.Summary().DataPoints().At(l).SetStartTimestamp(nanosec)
+					}
+				}
+			}
+		}
+	}
+}
+
+func sortMetricDataPointSlices(ms pmetric.Metrics) {
+	for i := 0; i < ms.ResourceMetrics().Len(); i++ {
+		for j := 0; j < ms.ResourceMetrics().At(i).ScopeMetrics().Len(); j++ {
+			for k := 0; k < ms.ResourceMetrics().At(i).ScopeMetrics().At(j).Metrics().Len(); k++ {
+				m := ms.ResourceMetrics().At(i).ScopeMetrics().At(j).Metrics().At(k)
+				switch m.Type() {
+				case pmetric.MetricTypeGauge:
+					sortNumberDataPointSlice(m.Gauge().DataPoints())
+				case pmetric.MetricTypeSum:
+					sortNumberDataPointSlice(m.Sum().DataPoints())
+				case pmetric.MetricTypeHistogram:
+					sortHistogramDataPointSlice(m.Histogram().DataPoints())
+				case pmetric.MetricTypeExponentialHistogram:
+					sortExponentialHistogramDataPointSlice(m.ExponentialHistogram().DataPoints())
+				case pmetric.MetricTypeSummary:
+					sortSummaryDataPointSlice(m.Summary().DataPoints())
+				}
+
+			}
+		}
+	}
+}
+
+func sortNumberDataPointSlice(ndps pmetric.NumberDataPointSlice) {
+	ndps.Sort(func(a, b pmetric.NumberDataPoint) bool {
+		aTime := a.Timestamp().String()
+		bTime := b.Timestamp().String()
```

Comment:
I think we need to incorporate more information in order to make this a deterministic sort. Specifically, we need to include attributes in the sort order.

For example, the timestamps on these data point could change and it would cause us to sort differently:
```
datapoints:
  - attributes:
    -  foo: bar
    timestamp: 1000
  - attributes:
        hello: world
    timestamp: 1001 # if changed to 999, this would sort to first position
```

Instead, we should sort first based on attributes, then on timestamps.